src/Features/VisualBasic/Portable/ConvertLinq/VisualBasicConvertLinqQueryToLinqMethodProvider.vb

```diff
@@ -0,0 +1,234 @@
+                    If expression Is Nothing Then
+                        Return Nothing
+                    End If
+                Next
+
+                Return expression.WithTrailingTrivia(source.GetTrailingTrivia())
+            End Function
+
+            Private Function ProcessQueryClause(expression As ExpressionSyntax, queryClause As QueryClauseSyntax) As ExpressionSyntax
+                Select Case (queryClause.Kind())
+                    Case SyntaxKind.FromClause
+                        Return ProcessFromClause(expression, DirectCast(queryClause, FromClauseSyntax))
+                    Case SyntaxKind.DistinctClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Distinct))
+                    Case SyntaxKind.WhereClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Where), DirectCast(queryClause, WhereClauseSyntax).Condition)
+                    Case SyntaxKind.SkipClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Skip), DirectCast(queryClause, PartitionClauseSyntax).Count)
+                    Case SyntaxKind.TakeClause
+                        Return CreateInvocationExpression(expression, NameOf(Enumerable.Take), DirectCast(queryClause, PartitionClauseSyntax).Count)
+                    Case SyntaxKind.OrderByClause
+                        Return ProcessOrderByClause(expression, DirectCast(queryClause, OrderByClauseSyntax))
+                    Case SyntaxKind.SelectClause
+                        Return ProcessSelectClause(expression, DirectCast(queryClause, SelectClauseSyntax))
+                    Case SyntaxKind.GroupJoinClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.SimpleJoinClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.LetClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.AggregateClause
+                        Return Nothing ' Too complicated for query method syntax.
+                    Case SyntaxKind.GroupByClause
+                        Return Nothing ' Group by in query and in method are not 100% equivalent. Consdier support of some group by in the future.
+                    Case Else
+                        Return Nothing
+                End Select
+            End Function
+
+            Private Function ProcessFromClause(parentExpression As ExpressionSyntax, fromClause As FromClauseSyntax) As ExpressionSyntax
+                Dim expression = parentExpression
+                Dim variables = fromClause.Variables
+                If variables.Count > 1 Then
+                    Return Nothing ' Do not support more than 1 variable. It complicates query methods.
+                End If
+
+                Dim variable = variables.First()
```

Comment:
>Dim [](start = 16, length = 3)

I understand this is IDE code and whatnot. However, if you are interested in getting as much value as possible from me reviewing the code, please no Dims without types. I am not a compiler and don't know how to do a type inference without getting a headache, which I will try to avoid. My recommendation would be to follow rules used by compiler team - the type must be spelled out unless it is clearly spelled out in the initializer.